File: AnandERP/AERP.DTO/EntityDTO/CCRM/CCRMComplaintLoggingMaster.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AERP.Base.DTO;

namespace AERP.DTO
{
   public class CCRMComplaintLoggingMaster :BaseDTO
    {
        public Int32 ID { get; set; }
        public Int32 UserID { get; set; }
        public string CallDate { get; set; }
        public string CallTime { get; set; }
        public string CallTktNo { get; set; }
        public string SerialNo { get; set; }
        public string MIFName { get; set; }
        public string CompanyCallDate { get; set; }
        public string CompanyCallTime { get; set; }
        public string CompanyCallNo { get; set; }
        public string CustomerName { get; set; }
        public Int32 CustomertID { get; set; }
        public string Phoneno { get; set; }
        public string EmailID { get; set; }
        public string SymptomCode { get; set; }
        public string SymptomTitle { get; set; }
        public string CallerName { get; set; }
        public string CallerPh { get; set; }
        public string ComPlaint { get; set; }
        public byte MachineStatus { get; set; }
        public byte Priority { get; set; }
        public Int32 A4Mono { get; set; }
        public Int32 A4Col { get; set; }
        public Int32 A3Mono { get; set; }
        public Int32 A3Col { get; set; }
        public decimal CallCharges { get; set; }
        public bool TeleSolution { get; set; }
        public bool CustApproval { get; set; }
        public string Remarks { get; set; }
        public string SplInstructions { get; set; }
        public bool SSSApproval { get; set; }
        public string MCAddress { get; set; }
        public string ContactPerson { get; set; }
        public Int32 ContactPersonID { get; set; }
        public string ModelNo { get; set; }
        public string ContractType { get; set; }
        public Int32 ContractTypeID { get; set; }
        public string ContractNo { get; set; }
        public Int32 ContractTypeId { get; set; }
        public string ContOpDate { get; set; }
        public string ContClDate { get; set; }
        public string SerEnggName { get; set; }
        public string EnggMobNo { get; set; }
        public string SplRemarks { get; set; }
        public string CallType { get; set; }
        public Int32 CallTypeID { get; set; }
        public Int32 EngineerID { get; set; }
        public Int32 SymptomID { get; set; }
        public string KeyOperator { get; set; }
        public byte CallStatus { get; set; }
        public Nullable<bool> IsDeleted { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<System.DateTime> CreatedDate { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public Nullable<int> DeletedBy { get; set; }
        public Nullable<System.DateTime> DeletedDate { get; set; }
        public string errorMessage { get; set; }
        public string ItemDescription { get; set; }
        public Int32 KeyOperatorID { get; set; }
        public string Solution { get; set; }
        public string CallTypeName { get; set; }
        public string CCRMComplaintLoggingMasterModel { get; set; }
        public string KeyOperatorName { get; set; }
        public string AreaPatchName { get; set; }
        public Int32 MIFID { get; set; }
        public Int32 ComplaintSrNo { get; set; }
        public bool Allotment { get; set; }
        public bool CallerFlag { get; set; }
        public Int16 AdminRoleMasterID { get; set; }
        public string CentreCode { get; set; }
        public string RightName { get; set; }
        public string EmployeeCode { get; set; }
        public string UserName { get; set; }
        public string EmployeeName { get; set; }
        public Int32 EmployeeID { get; set; }
        public Int32 CallerID { get; set; }
        public string VersionNumber { get; set; }
        public Nullable<System.DateTime> LastSyncDate { get; set; }
        public string SyncType
        {
            get; set;
        }
        public string Entity
        {
            get; set;
        }

        public string BrokenReason
        {
            get; set;
        } 
        public byte TrackType
        {
            get;
            set;
        }
        public string DeviceToken
        {
            get;
            set;
        }
        public string ComplaintNumberString
        {
            get;
            set;
        }
        public string JSON
        {
            get;
            set;
        }
        public decimal Latitude
        {
            get;
            set;
        }

        public decimal Longitude
        {
            get;
            set;
        }
        public Int32 ComplaintCountID { get; set; }
        public Int32 ComplaintCount { get; set; }

        public string ContractClosingDate
        {
            get;
            set;
        }
    }
}

File: AnandERP/AERP.Web.UI/Scripts/AERP_Scripts/CCRM/CCRMComplaintLoggingMaster.js
//this class contain methods related to nationality functionality
var CCRMComplaintLoggingMaster = {
    //Member variables
    ActionName: null,
    //Class intialisation method
    Initialize: function () {
        //organisationStudyCentre.loadData();

        CCRMComplaintLoggingMaster.constructor();
        //CCRMComplaintLoggingMaster.initializeValidation();
    },
    //Attach all event of page
    constructor: function () {
        $("#reset").click(function () {

            $("input,textarea").not(':reset, :submit, :hidden, :button, :checkbox,:radio').val("");
            $('input:checkbox,input:radio').removeAttr('checked');
            $('#SelectedRegionID').focus();
            $('#SelectedRegionID').val('');
        });

        // Create new record
        $('#CreateCCRMComplaintLoggingMasterRecord').on("click", function () {
            debugger;
            CCRMComplaintLoggingMaster.ActionName = "Create";
            CCRMComplaintLoggingMaster.AjaxCallCCRMComplaintLoggingMaster();
        });

        $('#EditCCRMComplaintLoggingMasterRecord').on("click", function () {

            CCRMComplaintLoggingMaster.ActionName = "Edit";
            CCRMComplaintLoggingMaster.AjaxCallCCRMComplaintLoggingMaster();
        });

        $('#DeleteCCRMComplaintLoggingMasterRecord').on("click", function () {

            CCRMComplaintLoggingMaster.ActionName = "Delete";
            CCRMComplaintLoggingMaster.AjaxCallCCRMComplaintLoggingMaster();
        });
        //$('#ActionTitle').on("keydown", function (e) {
        //    AERPValidation.AllowCharacterOnly(e);
        //});
        //$('#FeedbackPoints').on("keydown", function (e) {
        //    AERPValidation.AllowNumbersOnly(e);
        //});
        $('#CallDate').datetimepicker({
            format: 'DD MMM YYYY hh:mm',
          //  format: 'YYYY-MM-DD hh:mm',
             //allowInputToggle: true
            // maxDate: moment(),
            //language: 'en',
            //pick12HourFormat: true
          //  inline: true,
           // sideBySide: true
        });
       
        $('#CompanyCallDate').datetimepicker({
            format: 'DD MMM YYYY',
            // maxDate: moment(),

        });
        $('#ContOpDate').datetimepicker({
            format: 'DD MMM YYYY',
            // maxDate: moment(),

        });
        $('#ContClDate').datetimepicker({
            format: 'DD MMM YYYY',
            // maxDate: moment(),

        });
       
      
        $("#UserSearch").keyup(function () {
            oTable = $("#myDataTable").dataTable();
            oTable.fnFilter(this.value);
        });

        $("#searchBtn").click(function () {
            $("#UserSearch").focus();
        });


        $("#showrecord").change(function () {
            var showRecord = $("#showrecord").val();
            $("select[name*='myDataTable_length']").val(showRecord);
            $("select[name*='myDataTable_length']").change();
        });

        //$(".ajax").colorbox();
        InitAnimatedBorder();
        CloseAlert();

    },
    //LoadList method is used to load List page
    LoadList: function () {

        $.ajax({
            cache: false,
            type: "POST",
            dataType: "html",
            url: '/CCRMComplaintLoggingMaster/List',
            success: function (data) {
                //Rebind Grid Data
                $('#ListViewModel').html(data);
            }
        });
    },
    //ReloadList method is used to load List page
    ReloadList: function (message, colorCode, actionMode) {

        $.ajax(
        {
            cache: false,
            type: "POST",
            dataType: "html",
            data: { actionMode: actionMode },
            url: '/CCRMComplaintLoggingMaster/List',
            success: function (data) {
                //Rebind Grid Data
                $("#ListViewModel").empty().append(data);
                //twitter type notification
                //$('#SuccessMessage').html(message);
                //$('#SuccessMessage').delay(400).slideDown(400).delay(15000).slideUp(400).css('background-color', colorCode);
                notify(message, colorCode);
            }
        });
    },


    //Fire ajax call to insert update and delete record
    AjaxCallCCRMComplaintLoggingMaster: function () {
        var CCRMComplaintLoggingMasterData = null;
        if (CCRMComplaintLoggingMaster.ActionName == "Create") {
            $("#FormCreateCCRMComplaintLoggingMaster").validate();
            if ($("#FormCreateCCRMComplaintLoggingMaster").valid()) {
                CCRMComplaintLoggingMasterData = null;
                CCRMComplaintLoggingMasterData = CCRMComplaintLoggingMaster.GetCCRMComplaintLoggingMaster();
                ajaxRequest.makeRequest("/CCRMComplaintLoggingMaster/Create", "POST", CCRMComplaintLoggingMasterData, CCRMComplaintLoggingMaster.Success);
            }
        }
        else if (CCRMComplaintLoggingMaster.ActionName == "Edit") {
            $("#FormEditCCRMComplaintLoggingMaster").validate();
            if ($("#FormEditCCRMComplaintLoggingMaster").valid()) {
                CCRMComplaintLoggingMasterData = null;
                CCRMComplaintLoggingMasterData = CCRMComplaintLoggingMaster.GetCCRMComplaintLoggingMaster();
                ajaxRequest.makeRequest("/CCRMComplaintLoggingMaster/Edit", "POST", CCRMComplaintLoggingMasterData, CCRMComplaintLoggingMaster.Success);
            }
        }
        else if (CCRMComplaintLoggingMaster.ActionName == "Delete") {
            CCRMComplaintLoggingMasterData = null;
            //$("#FormCreateCCRMComplaintLoggingMaster").validate();
            CCRMComplaintLoggingMasterData = CCRMComplaintLoggingMaster.GetCCRMComplaintLoggingMaster();

            ajaxRequest.makeRequest("/CCRMComplaintLoggingMaster/Delete", "POST", CCRMComplaintLoggingMasterData, CCRMComplaintLoggingMaster.Success);

        }
    },
    //Get properties data from the Create, Update and Delete page
    GetCCRMComplaintLoggingMaster: function () {

        var Data = {
        };
        if (CCRMComplaintLoggingMaster.ActionName == "Create" || CCRMComplaintLoggingMaster.ActionName == "Edit") {
            debugger;
            Data.ID = $('#ID').val();

            Data.CallDate = $('#CallDate').val();
            Data.CallTktNo = $('#CallTktNo').val();
            Data.CallTypeID = $('#CallTypeID').val();
            Data.SerialNo = $('#SerialNo').val();
            Data.MIFName = $('#MIFName').val();
            Data.CompanyCallDate = $('#CompanyCallDate').val();
            Data.CompanyCallNo = $('#CompanyCallNo').val();
            Data.CallerName = $('#CallerName').val();
            Data.CallerPh = $('#CallerPh').val();
            Data.EmailID = $('#EmailID').val();
            Data.SymptomID = $('#SymptomID').val();
            Data.SymptomCode = $('#SymptomCode').val();
            Data.SymptomTitle = $('#SymptomTitle').val();
            Data.ComPlaint = $('#ComPlaint').val();
            Data.MachineStatus = $('#MachineStatus').val();
            Data.Priority = $('#Priority').val();
            Data.A4Mono = $('#A4Mono').val();
            Data.A4Col = $('#A4Col').val();
            Data.A3Mono = $('#A3Mono').val();
            Data.A3Col = $('#A3Col').val();
            Data.CallCharges = $('#CallCharges').val();
            Data.CustApproval = $('input[id=CustApproval]:checked').val() ? true : false;
            Data.TeleSolution = $('input[id=TeleSolution]:checked').val() ? true : false;
            Data.SSSApproval = $('input[id=SSSApproval]:checked').val() ? true : false;
        
            Data.SplRemarks = $('#SplRemarks').val();
            Data.SplInstructions = $('#SplInstructions').val();
            Data.MCAddress = $('#MCAddress').val();
            Data.CustomertID = $('#CustomertID').val();
            Data.CustomerName = $('#CustomerName').val();
            Data.KeyOperatorID = $('#KeyOperatorID').val();
            Data.KeyOperator = $('#KeyOperator').val();
            Data.Phoneno = $('#Phoneno').val();
            Data.ModelNo = $('#ModelNo').val();
            Data.ContractTypeId = $('#ContractTypeId').val();
            Data.ContractType = $('#ContractType').val();
            Data.ContractNo = $('#ContractNo').val();
            Data.ContOpDate = $('#ContOpDate').val();
            Data.ContClDate = $('#ContClDate').val();
            Data.EngineerID = $('#EngineerID').val();
            Data.EnggMobNo = $('#EnggMobNo').val();
            Data.ItemDescription = $('#ItemDescription').val();
            Data.Solution = $('#Solution').val();
            Data.ComplaintSrNo = $('#ComplaintSrNo').val();
            Data.MIFID = $('#MIFID').val();
            Data.Allotment = $('input[id=Allotment]:checked').val() ? true : false;
            Data.CallStatus = $('#CallStatus').val();
        }
        else if (CCRMComplaintLoggingMaster.ActionName == "Delete") {
            Data.ID = $('#ID').val();
        }
        return Data;
    },


    //this is used to for showing successfully record creation message and reload the list view
    Success: function (data) {
        var splitData = data.split(',');
        if (data != null) {
            //parent.$.colorbox.close();
            $.magnificPopup.close();

            CCRMComplaintLoggingMaster.ReloadList(splitData[0], splitData[1], splitData[2]);
        } else {
            //parent.$.colorbox.close();
            $.magnificPopup.close();

            CCRMComplaintLoggingMaster.ReloadList(splitData[0], splitData[1], splitData[2]);
        }

    },
    //this is used to for showing successfully record updation message and reload the list view
    //editSuccess: function (data) {

    //  
    //  
    //    if (data == "True") {

    //        parent.$.colorbox.close();
    //        var actionMode = "1";
    //        CCRMComplaintLoggingMaster.ReloadList("Record Updated Sucessfully.", actionMode);
    //        //  alert("Record Created Sucessfully.");

    //    } else {
    //        parent.$.colorbox.close();
    //        // alert("Record Not Updated Sucessfully. Please Try Again..");
    //    }

    //},
    ////this is used to for showing successfully record deletion message and reload the list view
    //deleteSuccess: function (data) {

    //  
    //    if (data == "True") {

    //        parent.$.colorbox.close();
    //        CCRMComplaintLoggingMaster.ReloadList("Record Deleted Sucessfully.");
    //        //  alert("Record Deleted Sucessfully.");

    //    } else {
    //        parent.$.colorbox.close();
    //        // alert("Record Not Deleted Sucessfully. Please Try Again..");
    //    }


    //},
};

